### ISSUE TITLE
'Server not reachable'; (re)open if Kõnele default server not working

### ISSUE THREAD

**Issue by mrts** [Kaljurand/K6nele#63]
Hello! 

Many-many thanks once again for this great app! I find it indispensable and use it a lot. However, since yesterday I get the 'server not reachable' error message when I try to dictate something.

**Comment by alumae** [comment]
Seems that TTÜ network maintenance has broken something. Should be fixed in a few days if not sooner.

**Comment by mrts** [comment]
Alright, fingers crossed that it's an easy fix!

**Comment by alumae** [comment]
Should be fixed now.

**Comment by mrts** [comment]
Yep, works again, many thanks!

**Comment by Te-M** [comment]
I haven't actually been able to try this app out because a webservice used for my configuration (http://bark.phon.ioc.ee/speech-api/v1) has been down since before the new year.  As you can see it throws a "503 Service Temporarily Unavailable" which I thought might be related but it's certainly not fixed.  Hope that was helpful, thanks for your time!

**Comment by Kaljurand** [comment]
@Te-M The service that you refer to (labeled as "grammar support" in the Kõnele UI) will probably be deprecated, because the corresponding server is difficult to maintain. For most tasks it is better use the "fast recognition" service (with the URL ws://bark.phon.ioc.ee:82/dev/duplex-speech-api/ws/speech) instead. The only feature of the "grammar support" service that is not provided by the "fast recognition" service is support for uploading and using GF or JSGF grammars (to replace the LM). Are you relying on this feature?

**Comment by Te-M** [comment]
Hmm, probably just an outdated tutorial.  Thanks for your help!

**Comment by Kaljurand** [comment]
Which tutorial?

**Comment by mrts** [comment]
Hello! Same error again, I hope it is an easy, hassle free fix this time too.

Heartfelt thanks for your good work once again!

**Comment by Kaljurand** [comment]
Thanks for reporting. I can confirm that the server is down at least since yesterday evening. The issue is being investigated. For some cases the work-around is to set up and use your own server, see https://hub.docker.com/r/alumae/docker-konele/ and http://kaljurand.github.io/K6nele/docs/et/user_guide.html#tuvastusserver-koduv%C3%B5rgus

**Comment by sheminnyoka** [comment]
Is the server down issue will be solved soon

**Comment by alumae** [comment]
Works again.

**Comment by mrts** [comment]
I can confirm that it works - thank you!

**Comment by mrts** [comment]
I see that you already noticed that the service is down, hope it is an easy fix yet again!

**Comment by mrts** [comment]
Works again, thank you for a quick fix!

**Comment by mrts** [comment]
Hm, seems that the service is down again after such a short time. Is there a systemic issue perhaps?

**Comment by BalooRJ1** [comment]
Hello,

The default server for English is once again no longer working...I am getting the same error "Server not reachable." Is there any way to fix this? Thank you.

**Comment by Kaljurand** [comment]
@BalooRJ1 sorry for the delay, can you clarify what do you mean by "the default server for English". As far as I can see, Kõnele does not come with any server for English...

**Comment by BalooRJ1** [comment]
the default server URL ws://bark.phon.ioc.ee:82/dev/duplex-speech-api/ws/speech and also http://bark.phon.ioc.ee/speech-api/v1/recognize bring back errors saying [server not reachable.] Now I'm not sure if this is because I have it set to English as the default language or if the servers actually are not functioning at all.

**Comment by khlsvr** [comment]
The current situation on the english server is same as in the previous message I guess?   

By the way, would it be difficult to implement the heavy lifting, the server, on the same device (mobile phone) where the client itself is or are the system requirements too high for that?

**Comment by Kaljurand** [comment]
Answers to your questions:

- yes, http://bark.phon.ioc.ee/speech-api/v1/recognize is not maintained anymore (@alumae would know if it ever comes back in some form)

- I'm not aware of any public/open/free Finnish server (but I haven't looked for this either)

- talk to the makers of vosk-android-demo, all they have to do is implement the SpeechRecognizer API (as described in 
https://github.com/Kaljurand/K6nele/discussions/38#discussioncomment-188542). Something similar was requested in https://github.com/microg/GmsCore/issues/1164

**Comment by nshmyrev** [comment]
> talk to the makers of vosk-android-demo, all they have to do is implement the SpeechRecognizer API (as described in

I was hoping for your help @Kaljurand  ;)

**Comment by Kaljurand** [comment]
@nshmyrev what help would be needed? It's just about implementing a standard Android API (+ ideally a few Kõnele-specific EXTRAs). In general, Kõnele is agnostic to what backend is being used. See also the related request #78

**Comment by nshmyrev** [comment]
Nevermind, we'll get some student to do this ;)

**Comment by khlsvr** [comment]
I'm happy to see that this is being considered! 

So what I understood is that the plan is to for example fork the K6nele-service project and then replace the code in it that currently connects to a remote kaldi-gstreamer-server to the code of the vosk-android-demo or perhaps some parts of it adding some API functionality to be able to "attach" it to this main k6nele app?

**Comment by Kaljurand** [comment]
@khlsvr yes, that would be the way to do it. The app needs to implement the service and declare it in the manifest (see e.g. https://github.com/Kaljurand/K6nele-service/blob/8a5ad02e19eb943736e02b44131a2c2bb34351b0/app/src/main/AndroidManifest.xml#L75), then Kõnele will find it.

**Comment by nshmyrev** [comment]
Also https://github.com/Kaljurand/K6nele/issues/68

**Comment by mrts** [comment]
Getting 'server does not work' on Android 12, is the service down? 
![Screenshot_20230628-125256](https://github.com/Kaljurand/K6nele/assets/44880/1f7e9f69-0d24-470c-bb1b-7e0a542ed7b5)

**Comment by Kaljurand** [comment]
Issue has been resolved

**Comment by mrts** [comment]
Happy new year and many thanks for this great service once again! Getting "serveriga puudub ühendus" again.

**Comment by Kaljurand** [comment]
Meanwhile fixed.

**Comment by mrts** [comment]
Getting erratic results since some time already , sometimes the application seems to infinitely waiting for server response, sometimes there are "transcription failed" errors. It used to work really well not long ago. Can this be caused by server issues?

Thanks in advance for looking into this!

**Comment by Kaljurand** [comment]
I can confirm based on my Pixel 6 with the recent versions of Android (currently AP41.240925.009) that the app behavour has recently deteriorated. I will have to do more testing with older the phones and other servers to understand this better.

@alumae can you comment on the status of the default server.

**Comment by Kaljurand** [comment]
Closing as server seems to be working now. Tested on K6nele v1.9.17 on Pixel 6 with Android v16 QPR1 Beta 1 (BP31.250502.008).

**Comment by mrts** [comment]
Thanks, confirming it is working well again. Missed it a lot, had to use Google voice input meanwhile which is much more erratic - it feels great to use Kõnele again!

**Comment by Kaljurand** [comment]
Pinging @alumae for possible updates on the server status.

**Comment by kaidoloor** [comment]
I get [server not reachable] and exit. Device: Samsung Z Fold5 (SM-F946B/DS), Andorid 16, One UI 8.0. This is irrespective of which Estonian 'recognition language & services' I select (even Claude or 'Speech recognition and synthesis).
